src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/alerts/AlertsByPrisonerMigrationService.kt

```diff
@@ -0,0 +1,95 @@
+import org.springframework.data.domain.PageImpl
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.data.MigrationContext
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.helpers.trackEvent
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomismappings.model.AlertMappingDto
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomissync.model.PrisonerId
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.AuditService
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.MigrationHistoryService
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.MigrationQueueService
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.MigrationService
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service.MigrationType
+
+@Service
+class AlertsByPrisonerMigrationService(
+  queueService: MigrationQueueService,
+  private val alertsNomisService: AlertsNomisApiService,
+  alertsMappingService: AlertsByPrisonerMigrationMappingApiService,
+  private val alertsDpsService: AlertsDpsApiService,
+  migrationHistoryService: MigrationHistoryService,
+  telemetryClient: TelemetryClient,
+  auditService: AuditService,
+  @Value("\${alerts.page.size:1000}") pageSize: Long,
+  @Value("\${alerts.complete-check.delay-seconds}") completeCheckDelaySeconds: Int,
+  @Value("\${alerts.complete-check.count}") completeCheckCount: Int,
+) : MigrationService<AlertsMigrationFilter, PrisonerId, AlertsForPrisonerResponse, List<AlertMappingDto>>(
+  queueService = queueService,
+  auditService = auditService,
+  migrationHistoryService = migrationHistoryService,
+  mappingService = alertsMappingService,
+  telemetryClient = telemetryClient,
+  migrationType = MigrationType.ALERTS,
+  pageSize = pageSize,
+  completeCheckDelaySeconds = completeCheckDelaySeconds,
+  completeCheckCount = completeCheckCount,
+) {
+  private companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  override suspend fun getIds(
+    migrationFilter: AlertsMigrationFilter,
+    pageSize: Long,
+    pageNumber: Long,
+  ): PageImpl<PrisonerId> {
+    return alertsNomisService.getPrisonerIds(
+      pageNumber = pageNumber,
+      pageSize = pageSize,
+    )
+  }
+
+  override suspend fun migrateNomisEntity(context: MigrationContext<PrisonerId>) {
+    log.info("attempting to migrate ${context.body}")
+    val latestNomisBookingId = context.body.bookingId
+    val offenderNo = context.body.offenderNo
+    val status = context.body.status
+
+    val nomisAlerts = alertsNomisService.getAlertsToMigrate(offenderNo)
+    val activeAlertsFromPreviousBookings = nomisAlerts.previousBookingsAlerts.filter { it.isActive }
+    alertsDpsService.migrateAlerts(nomisAlerts.latestBookingAlerts.map { it.toDPSMigratedAlert(context.body.offenderNo) })?.also {
+      // TODO create mappings here
```

Comment:
really doesn't do much other than gets each prisoners alerts and logs what we would do